tasks_week3_omer_gertler1/include/Animal.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+//#include <string>
+//#include <ostream>
+#include "Location.h"
+
+using std::string;
+//using std::ostream;
+
+class Animal
+{
+protected:
+
+private:
+	string name = "";
+	string type = "";
+	Location loc;
+	bool isMoving = true;
+
+public:
+	Animal();
+	Animal(const std::string& name, const Location& loc);
+
+	void printDetails() const;
+	char getInitial() const;
+	Location getLocation() const;
+	virtual void step() = 0;
+	void stop();
```

Comment:
no use of virtual 
check this
https://stackoverflow.com/questions/2391679/why-do-we-need-virtual-functions-in-c